### PR TITLE
Fix V3 types

### PR DESCRIPTION
**Pull request by szg251**
Found some bugs while testing with LambdaBuffers

### CODE REVIEW

File: plutus-ledger-api/src/v3/transaction.rs [hunk 1]
```diff
@@ -26,26 +43,210 @@ use crate::{
...
+
+/////////////////////
+// TransactionHash //
+/////////////////////
+
+/// 32-bytes blake2b256 hash of a transaction body.
+///
+/// Also known as Transaction ID or `TxID`.
+/// Note: Plutus docs might incorrectly state that it uses SHA256.
+/// V3 TransactionHash uses a more efficient Plutus Data encoding
```
Comment (jaredponn) [inline plutus-ledger-api/src/v3/transaction.rs:59]:
```suggestion
/// V3 TransactionHash uses a more efficient Plutus Data encoding.
/// See <https://github.com/IntersectMBO/plutus/blob/1.38.0.0/plutus-ledger-api/src/PlutusLedgerApi/V3/Tx.hs#L37-L54> for details.
```

Might be helpful to include a reference -- up to you in the end!

File: plutus-ledger-api/src/v3/transaction.rs [hunk 2]
```diff
@@ -26,26 +43,210 @@ use crate::{
...
+}
+
+//////////////////////
+// TransactionInput //
+//////////////////////
+
+/// An input of a transaction
+///
+/// Also know as `TxOutRef` from Plutus, this identifies a UTxO by its transacton hash and index
+/// inside the transaction
```
Comment (jaredponn) [inline plutus-ledger-api/src/v3/transaction.rs:131]:
```suggestion
/// inside the transaction
/// See <https://github.com/IntersectMBO/plutus/blob/1.38.0.0/plutus-ledger-api/src/PlutusLedgerApi/V3/Tx.hs#L62-L73> for details.
```

**Review by jaredponn — APPROVED** [review]
Amazing!

File: plutus-ledger-api/src/v3/transaction.rs (hunk 1 above)
Comment (szg251) [reply plutus-ledger-api/src/v3/transaction.rs:59]: good idea, i'll add this to a follow up issue (I want to add references to all types, which will be a bit of work)

**Pull request merged**